### PR TITLE
(For David) Creates SiftImpl class, removes instance API, adds resume hook

### PR DESCRIPTION
**Pull request by garylee1**
This PR does several things:

- Moves instance logic out of Sift and into SiftImpl
- Adds onResume hook
- Adds new activity to test app
- Updates test apps to use new API

### CODE REVIEW

**Comment by garylee1** [comment]
TODO: update README.md and marketing site docs. Also, write how-to-upgrade doc to accompany new GitHub tag.

File: sift/src/main/java/siftscience/android/Sift.java [hunk 1]
```diff
@@ -3,57 +3,38 @@
...
 import com.google.gson.GsonBuilder;
-import com.google.gson.JsonSyntaxException;
 import com.google.gson.annotations.SerializedName;
-import com.sift.api.representations.MobileEventJson;
 
-/** The public API of the Sift client library. */
+/**
+ * The public API of the Sift client library.
+ */
 public class Sift {
```
Comment (ehrmann) [inline sift/src/main/java/siftscience/android/Sift.java]: Make `Sift` final.

Comment (ehrmann) [reply sift/src/main/java/siftscience/android/Sift.java]: And add a private, zero-arg constructor.

File: sift/src/main/java/siftscience/android/Sift.java [hunk 2]
```diff
@@ -3,57 +3,38 @@
...
     static final String SDK_VERSION = BuildConfig.VERSION_NAME;
-    static final String DEVICE_PROPERTIES_QUEUE_IDENTIFIER = "siftscience.android.device";
-    static final String APP_STATE_QUEUE_IDENTIFIER = "siftscience.android.app";
 
     static final Gson GSON = new GsonBuilder()
             .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
             .create();
 
-    private static Sift instance;
+    private static SiftImpl instance;
```
Comment (ehrmann) [inline sift/src/main/java/siftscience/android/Sift.java]: volatile

File: sift/src/main/java/siftscience/android/SiftImpl.java [hunk 3]
```diff
@@ -134,7 +133,11 @@ String archiveConfig() {
      * @return the configuration for the Sift instance
      */
     synchronized Sift.Config getConfig() {
```
Comment (ehrmann) [inline sift/src/main/java/siftscience/android/SiftImpl.java]: nit: if it's not getting memoized, the method doesn't need to be synchronized.

**Comment by ehrmann** [comment]
2a45944ecac5a3a6a03b9680e842a705108a04f4 lgtm

**Pull request merged**